alerter.c

```diff
@@ -1,19 +1,31 @@
 #include <stdio.h>
 #include <assert.h>
+#include "product_config.h"
+
+#ifdef PRODUCTION

```

Comment:
Using conditional compilation for separating production and test code may get complicated as the code size grows.
it is better to use interfaces or functions pointers or lambda functions (if available) to inject either production or test code based on the envt.